Branch-SDK/Branch-SDK/BranchShareLink.h

```diff
@@ -0,0 +1,112 @@
+@param completed    This parameter is YES if sharing completed successfully and the user did not cancel.
+@param error        This parameter contains any errors that occurred will attempting to share.
+*/
+- (void) branchShareLink:(BranchShareLink*_Nonnull)shareLink
+             didComplete:(BOOL)completed
+               withError:(NSError*_Nullable)error;
+@end
+
+#pragma mark - BranchShareLink
+
+/**
+The `BranchShareLink` class facilitates sharing Branch links using a `UIActivityViewController` 
+user experience.
+
+The `BranchShareLink` is a new class that is similar to but has more functionality than the old 
+`[BranchUniversalObject showShareSheetWithLinkProperties:...]` methods.
+
+The `BranchShareLink` is initialized with the `BranchUniversalObject` and `BranchLinkProperties` 
+objects that will be used to generate the Branch link.
+
+After the `BranchShareLink` object is created, set any configuration properties on the activity 
+sheet object, and then call `showFromViewController:anchor:` to show the activity sheet.
+
+A delegate on the BranchShareLink can further configure the share experience. For instance the link 
+parameters can be changed depending on the activity that the user selects.
+*/
+
+@interface BranchShareLink : NSObject
+
+/**
+Creates a BranchShareLink object.
+
+@oaram universalObject  The Branch Universal Object the will be shared.
+@param linkProperties   The link properties that the link will have.
+*/
+- (instancetype _Nullable) initWithUniversalObject:(BranchUniversalObject*_Nonnull)universalObject
+                                    linkProperties:(BranchLinkProperties*_Nonnull)linkProperties;
+
+
+///Returns an array of activity item providers, one for the Branch Universal Object,
+///one for the share text (if provided), and one for the shareObject (if provided).
+- (NSArray<UIActivityItemProvider*>*_Nonnull) activityItems;
+
+/**
+Presents a UIActivityViewController that shares the Branch link.
+
+@oaram viewController   The parent view controller from which to present the the activity sheet.
+@param anchor           The anchor point for the activity sheet. Used for iPad form factors.
+*/
+- (void) presentActivityViewControllerFromViewController:(UIViewController*_Nullable)viewController
+                                                  anchor:(UIBarButtonItem*_Nullable)anchor;
+
+///The title for the share sheet.
+@property (nonatomic, strong) NSString*_Nullable title;
+
+///Share text for the item.
+///This text can be changed later when the `branchShareSheetWillShare:` delegate method is called.
+@property (nonatomic, strong) NSString*_Nullable shareText;
+
+///An additional, user defined, non-typed, object to be shared.
+///This object can be changed later when the `branchShareSheetWillShare:` delegate method is called.
+@property (nonatomic, strong) id _Nullable shareObject;
+
+///The resulting Branch URL that was shared.
+@property (nonatomic, strong, readonly) NSURL*_Nullable shareURL;
```

Comment:
This is nice. I was always disappointed that I couldn't get the short URL back from showShareSheet. I wonder if this will be supported on Android. It would be nice to support this for RN and other-sdks in general.